### PR TITLE
Added functionality for deleting Wiser branches.

### PR DESCRIPTION
**Pull request by GilianJuice**
Corresponding Wiser PR: https://github.com/happy-geeks/wiser/pull/283

Asana ticket: https://app.asana.com/0/1201027711166952/1204924239600564

### CODE REVIEW

**Comment by MarkVanDijkHG** [comment]
@GilianJuice Because https://github.com/happy-geeks/wiser-task-scheduler/pull/95 has been approved and merged to develop there are now merge conflicts for this pull request. Can you please look at these?

**Comment by GilianJuice** [comment]
I merged develop into this branch and fixed the conflicts.

File: WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs [hunk 1]
```diff
@@ -530,17 +530,27 @@ FROM INFORMATION_SCHEMA.ROUTINES
 
                 // Rollback transaction if started
                 await databaseConnection.RollbackTransactionAsync(false);
+                logger.LogError(exception, $"Failed to create the branch '{settings.DatabaseName}'.");
```
Comment (MarkVanDijkHG) [inline WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs:533]: Please use the logService for logging. This ensures the log is written to the database and/or a Slack channel.

File: WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs [hunk 2]
```diff
@@ -530,17 +530,27 @@ FROM INFORMATION_SCHEMA.ROUTINES
...
                 {
-                    await databaseHelpersService.DropDatabaseAsync(branchDatabase);
+                    if (await databaseHelpersService.DatabaseExistsAsync(branchDatabase))
+                    {
+                        await databaseHelpersService.DropDatabaseAsync(branchDatabase);
+                    }
+                }
+                catch (Exception innerException)
+                {
+                    logger.LogError(innerException, "Failed to drop new branch database, after getting an error while trying to fill it with data.");
```
Comment (MarkVanDijkHG) [inline WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs:550]: Please use the logService for logging. This ensures the log is written to the database and/or a Slack channel.

File: WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs [hunk 3]
```diff
@@ -1513,6 +1524,7 @@ await using (var environmentCommand = branchConnection.CreateCommand())
             }
             catch (Exception exception)
             {
+                logger.LogError(exception, $"Failed to merge the branch '{settings.DatabaseName}'.");
```
Comment (MarkVanDijkHG) [inline WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs:1527]: Please use the logService for logging. This ensures the log is written to the database and/or a Slack channel.

File: WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs [hunk 4]
```diff
@@ -1622,6 +1634,100 @@ private static async Task FinishBranchActionAsync(int queueId, DataRow dataRowWi
...
+            {
+                // Delete the database, if it exists.
+                if (await databaseHelpersService.DatabaseExistsAsync(settings.DatabaseName))
+                {
+                    await databaseHelpersService.DropDatabaseAsync(settings.DatabaseName);
+                }
+            }
+            catch (Exception exception)
+            {
+                logger.LogError(exception, $"Failed to delete the branch '{settings.DatabaseName}'.");
```
Comment (MarkVanDijkHG) [inline WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs:1715]: Please use the logService for logging. This ensures the log is written to the database and/or a Slack channel.

File: WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs [hunk 5]
```diff
@@ -1622,6 +1634,100 @@ private static async Task FinishBranchActionAsync(int queueId, DataRow dataRowWi
...
+            // Set the finish time to the current datetime, so that we can see how long it took.
+            databaseConnection.AddParameter("queueId", queueId); // Set the queue ID again because if a data selector is used the parameters are cleared.
+            databaseConnection.AddParameter("now", DateTime.Now);
+            databaseConnection.AddParameter("error", error);
+            databaseConnection.AddParameter("success", String.IsNullOrWhiteSpace(error));
+            await databaseConnection.ExecuteAsync($"UPDATE {WiserTableNames.WiserBranchesQueue} SET finished_on = ?now, success = ?success, errors = ?error WHERE id = ?queueId");
+            result.Add("ErrorMessage", error);
+            result.Add("Success", String.IsNullOrWhiteSpace(error));
+            return result;
+
```
Comment (MarkVanDijkHG) [inline WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs:1728]: Please remove the redundant newline.

**Review by MarkVanDijkHG — CHANGES_REQUESTED** [review]

File: WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs (hunk 1 above)
Comment (GilianJuice) [reply WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs:533]: Done

File: WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs (hunk 2 above)
Comment (GilianJuice) [reply WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs:550]: Done

File: WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs (hunk 3 above)
Comment (GilianJuice) [reply WiserTaskScheduler/WiserTaskScheduler/Modules/Branches/Services/BranchQueueService.cs:1631]: Done

**Review by MarkVanDijkHG — APPROVED** [review]

**Pull request merged**